Dockerfile

```diff
@@ -0,0 +1,100 @@
+# configured for each vehicle by means of volume mapping configuration files and
+# networking mapping at run time
+#
+# Build Arguments:
+# SSH_PRIVATE_KEY - If the extra package repository to be used during the build requires
+#                   authentication, please pass in the necessary SSH private key in text
+#                   form here, most likely via "$(cat ~/.ssh/id_rsa)". This data is not 
+#                   present in the final output image. Default = none
+#
+# EXTRA_PACKAGES - The repo to checkout any additional packages from at build time. 
+#                  Default = none
+
+
+# /////////////////////////////////////////////////////////////////////////////
+# Stage 1 - Install the SSH private key and acquire the CARMA source as well as
+#           any extra packages
+# /////////////////////////////////////////////////////////////////////////////
+FROM ros:kinetic-ros-base AS source-code
+RUN apt-get update
+RUN apt-get install -y git ssh
+
+ARG SSH_PRIVATE_KEY
+ARG EXTRA_PACKAGES
+ENV EXTRA_PACKAGES ${EXTRA_PACKAGES}
+ARG EXTRA_PACKAGES_VERSION=master
+ENV EXTRA_PACKAGES_VERSION ${EXTRA_PACKAGES_VERSION}
+
+# Set up the SSH key for usage by git
+RUN mkdir /root/.ssh/
+RUN echo "${SSH_PRIVATE_KEY}" > /root/.ssh/id_rsa
+RUN chmod 600 /root/.ssh/id_rsa
+RUN touch /root/.ssh/known_hosts
+RUN ssh-keyscan github.com >> /root/.ssh/known_hosts
+
+# Acquire the software source
+COPY . /root/src/CARMAPlatform/
+RUN /root/src/CARMAPlatform/docker/checkout.sh
+
+
+# /////////////////////////////////////////////////////////////////////////////
+# Stage 2 - Build and install the software 
+# /////////////////////////////////////////////////////////////////////////////
+FROM ros:kinetic-ros-base AS install
+
+# Install necessary packages
+RUN apt-get update
+RUN apt-get install -y ros-kinetic-rosjava ros-kinetic-rosbridge-server
+RUN rosdep update
+
+# Copy the source files from the previous stage and build/install
+COPY --from=source-code /root/src /root/carma_ws/src
+RUN ~/carma_ws/src/CARMAPlatform/docker/install.sh
+
+# /////////////////////////////////////////////////////////////////////////////
+# Stage 3 - Finalize deployment
+# /////////////////////////////////////////////////////////////////////////////
+FROM ros:kinetic-ros-base
+
+# Add carma user
+ENV USERNAME carma
+RUN useradd -m $USERNAME && \
+        echo "$USERNAME:$USERNAME" | chpasswd && \
+        usermod --shell /bin/bash $USERNAME && \
+        usermod -aG sudo $USERNAME && \
+        mkdir -p /etc/sudoers.d && \
+        echo "$USERNAME ALL=(ALL) NOPASSWD:ALL" >> /etc/sudoers.d/$USERNAME && \
+        chmod 0440 /etc/sudoers.d/$USERNAME && \
+        usermod  --uid 1000 $USERNAME && \
+        groupmod --gid 1000 $USERNAME
+
+# Install necessary packages
+RUN apt-get update
+RUN apt-get install -y sudo ros-kinetic-rosjava ros-kinetic-rosbridge-server ros-kinetic-robot tmux apache2 php7.0 libapache2-mod-php7.0 vim nano less
+
```

Comment:
might be worth having a rosdep update call here as you do in line 66 above, since this new container has reinstalled rosjava, rosbridge, and robot

Reply:
I thought the `rosdep update` was the equivalent of `apt-get update`, eg just fetching a new registry. of available packages in repos. This stage of the build doesn't use `rosdep` to install anything so I'm not sure that's needed.